### PR TITLE
Fips test hooks

### PR DESCRIPTION
**Pull request by nquarton**

### CODE REVIEW

**Review by jlmahowa-amd — DISMISSED** [review]

File: test/tests/fips_test_suite/self_tests.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    // Wait for fatal error
+    hw.step_until(|m| m.soc_ifc().cptra_fw_error_fatal().read() != 0);
+
+    // Verify fatal code is correct
+    assert_eq!(
+        hw.soc_ifc().cptra_fw_error_fatal().read(),
+        u32::from(CaliptraError::ROM_KAT_SHA384_DIGEST_MISMATCH)
+    );
+
+    // TODO: Verify we cannot use the algorithm
```
Comment (nquarton) [inline test/tests/fips_test_suite/self_tests.rs:68]: All of these test TODOs will be completed in a later PR with these full test implementations and additional tests. These partial tests were just included to demonstrate the usage of the FIPS test hooks to the point where the correct error code is received from the injected error.

File: rom/dev/src/main.rs [hunk 2]
```diff
@@ -187,7 +188,11 @@ pub extern "C" fn rom_entry() -> ! {
         CfiCounter::corrupt();
     }
 
-    #[cfg(not(feature = "no-fmc"))]
+    // FIPS test hooks mode does not allow handoff to RT to prevent incorrect/accidental usage
```
Comment (mhatrevi) [inline rom/dev/src/main.rs:191]: NIT: Should be "handoff to FMC"

File: test/tests/fips_test_suite/README.md [hunk 3]
```diff
@@ -36,4 +36,8 @@ These can be enabled using the --features argument for rust like:
 
 ## Additional Environments
 
-Support for additional environments can be done by creating new implementations/interfaces for the HW model at hw-model/src. See model_fpga_realtime.rs as an example. This implementation needs to be able to access the APB bus, control the input signals to caliptra, and, if possible, control ROM.
\ No newline at end of file
+Support for additional environments can be done by creating new implementations/interfaces for the HW model at hw-model/src. See model_fpga_realtime.rs as an example. This implementation needs to be able to access the APB bus, control the input signals to caliptra, and, if possible, control ROM.
+
+## Test Hooks
+
+Certain tests require "hooks" into the ROM or FW to cause operation to deviate from the normal flow (ie. injecting errors or halting execution at specific points). This functionality is enabled using a build option called "fips-test-hooks". Then, the specific command codes are written and read from the DBG_MANUF_SERVICE_REG. The ROM/FW can respond back with a status code written to the same field if applicable. See command codes in drivers\src\fips_test_hooks.rs for more details.
```
Comment (mhatrevi) [inline test/tests/fips_test_suite/README.md:43]: Can we add the FIPS requirement (number etc.) here that needs the test hook change?

**Review by mhatrevi — CHANGES_REQUESTED** [review]

File: rom/dev/src/main.rs (hunk 2 above)
Comment (nquarton) [reply rom/dev/src/main.rs]: Good point, that is more accurate. This is changed now

File: test/tests/fips_test_suite/README.md (hunk 3 above)
Comment (nquarton) [reply test/tests/fips_test_suite/README.md:43]: No problem. Added that list with the test requirements that need this

**Review by mhatrevi — DISMISSED** [review]

**Review by mhatrevi — APPROVED** [review]

**Review by jlmahowa-amd — APPROVED** [review]

**Pull request merged**